core/app/models/spree/tax/output/item.rb

```diff
@@ -0,0 +1,10 @@
+module Spree
+  module Tax
+    module Output
+      class Item
+        include ActiveModel::Model
+        attr_accessor :id, :tax_rate, :amount, :included_in_price
```

Comment:
Shipping rates are much easier to avoid passing database IDs. Since there's no need to tie the rate to any existing database record afterwards. If we were looking to make an interface for returning a list of tax rates for an item I would entirely agree with you.

However, we're looking to return a list of calculated taxes for items on an order. Meaning we need to tie those values together in some concrete way. Sorting the items and using their indices, giving them a new alphanumeric identifier, or using the database IDs are all functionally equivalent. Using the database IDs is just the simplest and most robust approach.